### ISSUE TITLE
JwtBearerAuthentication problem with HttpContext.GetTokenAsync  Core2.1

### ISSUE THREAD

**Issue by Temoxa** [aspnet/Home#3186]
Hi!

I upgrade to version 2.1 and have problem with method HttpContext.GetTokenAsync()

Prev code in version 2.0 work is successful! 

`var task = Task.Run(async () => await context.GetTokenAsync("access_token"));`
 `task.Wait();`
 `return task.Result;`

In version 2.1 now does not work(((( result always in NULL 

Help me please ^^

P.S. Configure Startup.cs
`public void ConfigureServices(IServiceCollection services)`
   `     {`
            `services.AddAuthentication(options =>`
                `    {`
                        `options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;`
                        `options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;`
                   ` })`
                    `.AddJwtBearer(options =>`
                   ` {`
                        `options.RequireHttpsMetadata = false;`
                        `options.TokenValidationParameters = new TokenValidationParameters`
                       ` {`
                           ` ValidateIssuer = true,`
                            `ValidIssuer = AuthOptions.ISSUER,`
                           ` ValidateAudience = true,`
                            `ValidAudience = AuthOptions.AUDIENCE,`
                            `ValidateLifetime = true,`
                            `IssuerSigningKey = AuthOptions.GetSymmetricSecurityKey(),`
                           ` ValidateIssuerSigningKey = true,`
                        `};`
        `}`

**Comment by Tratcher** [comment]
Duplicate of https://github.com/aspnet/Security/issues/1765

**Comment by Tratcher** [comment]
Side note:
```
var task = Task.Run(async () => await context.GetTokenAsync("access_token"));
task.Wait();
return task.Result;
```
is a terrible way to use Tasks and await. If you must run it sync then
`return context.GetTokenAsync("access_token").Result;`

But even .Result should be avoided, make the whole code path async.
return await context.GetTokenAsync("access_token");

**Comment by Temoxa** [comment]
Okay, but solution in issue aspnet/Security#1765 not worked.....

`context.AuthenticateAsync(JwtBearer.Defaults).Ticket.Properties.GetToken("access_token");`

return NULL value

@Tratcher  help )

**Comment by Tratcher** [comment]
Read to the bottom.
https://github.com/aspnet/Security/issues/1765#issuecomment-391455986

**Comment by Temoxa** [comment]
@Tratcher  thanks!

One question, this problem will be fixed?

**Comment by Tratcher** [comment]
It has already been fixed for 2.2.
